transformer_engine/pytorch/tensor/mxfp8_tensor.py

```diff
@@ -346,7 +346,8 @@ def _set_data(self, tensor: torch.Tensor) -> None:
         casts to FP8.
 
         """
-
+        if isinstance(tensor, MXFP8TensorBase):
+            return
```

Comment:
This will result in situations where calling `mxfp8_tensor.data = bf16_tensor` will silently do nothing. Better to prevent this function from constructing a `MXFP8TensorBase`. We can do this by changing https://github.com/NVIDIA/TransformerEngine/blob/0d45556ccdb8e418f6eeb1a1e7440a1893c15320/transformer_engine/pytorch/tensor/mxfp8_tensor.py#L388
```python
self._quantizer.internal = False
self.data = self._quantizer.quantize(tensor)
```